src/App.tsx

```diff
@@ -52,13 +57,36 @@ declare global {
 }
 
 function App() {
-  const [theme, setTheme] = React.useState(Theme.Light);
-  const [showHeader, setShowHeader] = React.useState(true);
-  const [showSplotches, setShowSplotches] = React.useState(false);
-  const [showBacklink, setShowBacklink] = React.useState(false);
+  const [theme, setTheme] = useState(Theme.Light);
+  const [showHeader, setShowHeader] = useState(true);
+  const [showSplotches, setShowSplotches] = useState(false);
+  const [showBacklink, setShowBacklink] = useState(false);
   const location = useLocation();
   const { persistLastUrl, isInNHSApp } = useContext(AuthContext);
   const MuiTheme = useTheme();
+  const { setInMaintenanceMode, inMaintenanceMode } = useMaintenance();
+
+  const [{ error, loading }] = useAxiosFetch(
+    {
+      url: `${process.env.REACT_APP_BASE_API}/health`,
+      method: "GET",
+    },
+    {
+      useCache: false,
+      manual: false,
+    }
+  );
+
+  useEffect(() => {
```

Comment:
I think there is another pattern for configuring globals / cross cutting concerns.
This code doesn't have side effects directly it is just setting up some relationships between components, so useEffect doesn't seem to be the right choice.

Possibly do the configuration directly in `useMaintenance`.